autopilot/purepursuitwaypointfollower.cpp

```diff
@@ -44,7 +44,45 @@ void PurepursuitWaypointFollower::addWaypoint(const PosPoint &point)
 
 void PurepursuitWaypointFollower::addRoute(const QList<PosPoint> &route)
 {
-    mWaypointList.append(route);
+    if (route.isEmpty()) {
+        return;
+    }
+
+    if (!isActive()) {
+        mWaypointList.clear();
```

Comment:
This clear changes the original behavior and I am not sure whether this is OK in all cases. Is it OK to remove?